jaxopt/_src/projection.py

```diff
@@ -136,6 +136,60 @@ def projection_simplex(x: jnp.ndarray, value: float = 1.0) -> jnp.ndarray:
   return value * _projection_unit_simplex(x / value)
 
 
+@partial(jax.custom_jvp, nondiff_argnums=(1, 2, 3))
+def projection_simplex_k_sparse(x: jnp.ndarray, k: int, value: float = 1.0,
```

Comment:
I would call it `projection_sparse_simplex` as it's shorter. `max_nz` is probably more intelligible than `k`.